metrics/engine/engine.go

```diff
@@ -29,35 +51,42 @@ type MetricsEngine struct {
 	test           *lib.TestRunState
 	outputIngester *outputIngester
 
-	// These can be both top-level metrics or sub-metrics
-	metricsWithThresholds []*metrics.Metric
+	// they can be both top-level metrics or sub-metrics
+	//
+	// TODO: remove the tracked map using the sequence number
+	metricsWithThresholds map[string]metrics.Thresholds
+	trackedMetrics        map[string]*trackedMetric
```

Comment:
We need to investigate as a better and long-term solution the `metric.SequenceID` option